chained/dialer.go

```diff
@@ -188,16 +191,6 @@ func defaultDialOrigin(op *ops.Op, ctx context.Context, p *proxy, network, addr
 		return nil, err
 	}
 
-	var timeout time.Duration
-	if deadline, set := ctx.Deadline(); set {
-		conn.SetDeadline(deadline)
-		// Set timeout based on our given deadline, minus a 2 second fudge factor
-		timeout := time.Until(deadline) - 2*time.Second
-		if timeout < 0 {
-			log.Errorf("Not enough time left for server to dial upstream within %v, return errUpstream immediately", timeout)
-			return nil, errUpstream
```

Comment:
Ah, yeah, good to fix this!

Reply:
For those following along at home: this was the subject of a discussion on Slack yesterday.  We don't want to treat timeouts as upstream errors (where "upstream error" roughly translates to "not an issue with the proxy") as timeouts are often indicative of blocking (because the censor drops the connection in an on-path middle-box).